### DISCUSSION TITLE
is it possible to add valid SSL certificate

### DISCUSSION THREAD

**Discussion by Kailashlambe** [wardenenv/warden#733] [Help & Support]
Hi Team,

I have completed the Warden Magento 2 set up, and I wanted to use the valid SSL certificate instead of default created warden ssl certificate, is it possible to add valid certificate, please confirm?

**Comment by navarr** [comment]
You can generate a self-signed certificate for an environment by running `warden sign-certificate <domain>` e.g. `warden sign-certificate magento2.test`

**Comment by Kailashlambe** [comment]
Thanks @navarr for your feedback, actually we do have signed certificate from client and I wanted to use that certificate on local instance, is it possible to use such kind of third party SSL certificate with warden?

**Reply by bap14** [reply]
@Kailashlambe Warden uses Traefik under the hood, and they can be added via the Traefik configuration (see: https://doc.traefik.io/traefik/https/tls/ ).

To do this, you can place your SSL cert and key file in `${WARDEN_HOME}/ssl/certs/` (where `${WARDEN_HOME}` is generally `~/.warden`), reload Traefik (`warden svc up traefik && warden svc restart traefik`) and then visit the site. Traefik's restart should automatically pick up the certificate and start using it for the matched domain.
